### PR TITLE
Adding filters to settings

### PR DESCRIPTION
**Pull request by Mouados-byte**
# Roles:

image

# Sites:

image

# Services:

image

# Holidays:

image

# Changes

image

### CODE REVIEW

**Comment by Mouados-byte** [comment]
[Add Filters to all parameters tabs => we have lists without search filters](https://trello.com/c/ffWDxkFs/272-add-filters-to-all-parameters-tabs-we-have-lists-without-search-filters)

File: app/views/sites/index.html.erb [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
-    <div class="sm:flex-auto">
-      <h2><%= t("views.sites.title_sites") %></h2>
-    </div>
+  <div class="sm:flex-auto">
+    <h2><%= t("views.sites.title_sites") %></h2>
+  </div>
 
+  <div class="flex justify-between flex-col-reverse sm:flex-row mt-5">
+    <%= render layout: "shared/filter_dropdown", locals: { reset_path: @url, reset_params: {} } do %>
+      <%= search_form_for @q, method: :get, url: @url, data: { turbo_frame: "search-results", turbo_action: "advance", controller: "search", action: "input->search#submit" } do |f| %>
```
Comment (itsAndL) [inline app/views/sites/index.html.erb:10]: for sites you can do all the filters in one filter

File: app/views/roles/index.html.erb [hunk 2]
```diff
@@ -1,38 +1,45 @@
...
+    <h2><%= t("views.roles.roles_title") %></h2>
+  </div>
+
+  <div class="flex justify-between flex-col-reverse sm:flex-row mt-5">
+    <%= render layout: "shared/filter_dropdown", locals: { reset_path: @url, reset_params: {} } do %>
+      <%= search_form_for @q, method: :get, url: @url, data: { turbo_frame: "search-results", turbo_action: "advance", controller: "search", action: "input->search#submit" } do |f| %>
+        <div class="space-y-3 mx-2 my-3">
           <div>
-            <%= f.label :user_first_name_or_user_last_name_cont, class: "sr-only", placeholder: "Role" %>
+            <%= f.label :user_first_name_or_last_name_or_email_cont, class: "sr-only", placeholder:t("views.roles.search_by") %>
```
Comment (MehdiElrafi) [inline app/views/roles/index.html.erb:12]: why you add filter by email?
and I won't able to filter by last_name you need to user_first_name_or_user_last_name_cont!
also for selecting roles in filter you need to make it by current user role (if current user has role hr he won't able to filter by admins, check roles_names_to_select() method in role model)

File: app/views/services/index.html.erb [hunk 3]
```diff
@@ -1,8 +1,26 @@
 <%= render partial: 'settings/settings_header' %>
 
 <div class="mt-5">
-  <div class="sm:flex sm:justify-end">
-    <!-- `new service` button -->
+  <div class="flex justify-between flex-col-reverse sm:flex-row mt-5">
+    <%= render layout: "shared/filter_dropdown", locals: { reset_path: @url, reset_params: {} } do %>
+      <%= search_form_for @q, method: :get, url: @url, data: { turbo_frame: "search-results", turbo_action: "advance", controller: "search", action: "input->search#submit" } do |f| %>
```
Comment (itsAndL) [inline app/views/services/index.html.erb:6]: you can put the search directly in the index page

bo beed the filter dropdown, cause you filter by one

**Review by itsAndL — CHANGES_REQUESTED** [review]

File: app/views/roles/index.html.erb (hunk 2 above)
Comment (itsAndL) [inline app/views/roles/index.html.erb:12]: Filter for role already exist

File: app/views/sites/index.html.erb (hunk 1 above)
Comment (Mouados-byte) [reply app/views/sites/index.html.erb:10]: what about the code ?

File: app/views/services/index.html.erb (hunk 3 above)
Comment (Mouados-byte) [reply app/views/services/index.html.erb:5]: Done

File: app/views/roles/index.html.erb (hunk 2 above)
Comment (Mouados-byte) [reply app/views/roles/index.html.erb]: Done

**Review by MehdiElrafi — APPROVED** [review]

**Review by itsAndL — APPROVED** [review]

**Pull request merged**